File: Java/300-400/AR_346.java
import java.util.Scanner;

public class AR_346 {
    
    // Pre: true
    static int dedos(int[] v, int A) {
        int maximo =0;
        int a      =0;
        int b      =1;
        int aux    =1;
       
        while (b<v.length) {
            if (v[b]<=v[a]+A-1) {
                aux++;
                b++;
            }
            else {
                a++;
                maximo=Math.max(maximo, aux);
                aux--;
            }
        }

        maximo=Math.max(maximo, aux);

        return maximo;
    }
    // Pos: r = Sum k: 0 <= k < v.size(): v[k]

    static boolean solve(Scanner sc) {
        
        int N=sc.nextInt();
        int A=sc.nextInt();
        if (N==0 && A==0) return false;

        int[] v=new int[N];
        for (int i=0;i<N;i++) v[i]=sc.nextInt();

        // compute solution
        int r=dedos(v, A);

        // write solution
        System.out.println(r);

        return true;
    }

    // usar buffered reader/writer da memory limit exceeded
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        while (solve(sc));

        sc.close();
    }
}

File: C/300-400/AR_346.c
#include <stdio.h>
#include <stdlib.h>


int max(int a, int b) { return (a>b) ? a:b; }

// Pre: true
int dedos(const int* v, int size, int A) {
    int ret =0;
    int a      =0;
    int b      =1;
    int aux    =1;
   
    while (b<size) {
        if (v[b]<=v[a]+A-1) {
            aux++;
            b++;
        }
        else {
            a++;
            ret=max(ret, aux);
            aux--;
        }
    }

    ret=max(ret, aux);

    return ret;
}
// Pos: r = Sum k: 0 <= k < v.size(): v[k]

int solve() {
    int i;

    int N=0, A=0;
    scanf("%d %d", &N, &A);
    if (N==0 && A==0) return 0;

    int v[N];
    for (i=0;i<N;i++) scanf("%d", &v[i]);
    
    int r=dedos(v, N, A);

    printf("%d\n", r);

    return 1;
}

int main() {
    while (solve());

    return 0;
}